ingestion_server/ingestion_server/api.py

```diff
@@ -47,124 +47,170 @@ def _get_base_url(req):
+                # https://www.postgresql.org/docs/current/datatype-datetime.html
+                "since_date": {"type": "string"},
+                "index_suffix": {"type": "string"},
+                "alias": {"type": "string"},
+            },
+            "required": ["model", "action"],
+            "allOf": [
+                {
+                    "if": {
+                        "properties": {"action": {"const": TaskTypes.UPDATE_INDEX.name}}
+                    },
+                    "then": {"required": ["index_suffix", "since_date"]},
+                },
+                {
+                    "if": {
+                        "properties": {"action": {"const": TaskTypes.POINT_ALIAS.name}}
+                    },
+                    "then": {"required": ["index_suffix", "alias"]},
+                },
+            ],
+        }
+    )
+    def on_post(self, req, res):
         """
-        Validate an index creation task.
-        :return: None if valid else a string containing an error message.
+        Handles an incoming POST request. Schedules the specified task.
+
+        :param req: the incoming request
+        :param res: the appropriate response
         """
-        if request == b"":
-            return "Expected JSON request body but found nothing."
-        request = json.loads(request.decode("utf-8"))
-        if MODEL not in request:
-            return "No model supplied in request body."
-        if ACTION not in request:
-            return "No action supplied in request body."
-        if request[ACTION] not in [x.name for x in TaskTypes]:
-            return "Invalid action."
-        if request[ACTION] == TaskTypes.UPDATE_INDEX.name and SINCE_DATE not in request:
-            return "Received UPDATE request but no since_date."
-
-        return None
-
-    def on_post(self, req, resp):
-        """Create a task."""
-        raw_body = req.stream.read()
-        request_error = self._validate_create_task(raw_body)
-        if request_error:
-            logging.warning(f"Invalid request made. Reason: {request_error}")
-            resp.status = falcon.HTTP_400
-            resp.media = {"message": request_error}
-            return
-        body = json.loads(raw_body.decode("utf-8"))
-        model = body[MODEL]
-        action = body[ACTION]
-        callback_url = None
-        if CALLBACK_URL in body:
-            callback_url = body[CALLBACK_URL]
-        since_date = body[SINCE_DATE] if SINCE_DATE in body else None
-        task_id = str(uuid.uuid4())
-        # Inject shared memory
+
+        body = req.get_media()
+
+        # Generated fields
+        task_id = uuid.uuid4().hex  # no hyphens
+
+        # Required fields
+
+        model: MediaType = body[MODEL]
+        action = TaskTypes[body[ACTION]]
+
+        # Optional fields
+        callback_url = body.get("callback_url")
+        since_date = body.get("since_date")
+        index_suffix = body.get("index_suffix", task_id)
```

Comment:
I re-tested with #578 and I think I'm seeing what @krysal  is seeing -- `NotFoundError(404, 'index_not_found_exception', 'no such index [audio]', audio, index_or_alias)`.

I did a small amount of investigating and strangely, it looks like we're not getting the same `index_suffix` here for the two POSTs. In both cases `index_suffix` is in the request body, but I'm seeing a different uuid. Therefore the `POINT_ALIAS` task fails to find the index.

When I initially tested this I generated a uuid locally, then used `curl` commands and manually passed the same one. This all works! So while it's a strange problem, I don't think it's an issue with this code 🤔